### PR TITLE
Feature/devise

### PR DESCRIPTION
**Pull request by Ayokunnumi1**
These are the new implementations in this branch. Project done by myself and partner @lily4178993 

- Removed current_user method in ApplicationController,
- Install and setup devise
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<section class="log-in-section">
+  <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+    <div class="logIn-inputs">
+      <%= f.text_field :name, autofocus: true, placeholder: "Username", autocomplete: "current-name", class: "name-field" %>
```
Comment (TedLivist) [inline app/views/devise/sessions/new.html.erb]: - [ ] After signing up and confirming my email, when I try to log in I keep getting an invalid email or password error. Kindly note that users should be able to sign in with email and password, not username and password, so you need to update this text field should be changed to email field the attribute should be changed from `:name` to `:email`. The placeholder should also be `Email`, not `Username`.

  ```
  <%= f.email_field :email, autofocus: true, placeholder: "Email", ... %>
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Ayokunnumi1 and @lily4178993,

Good job so far!
✔️ Integration of Devise to application
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (lily4178993) [reply app/views/devise/sessions/new.html.erb]: Hi @TedLivist , Did you confirm your email before sign in?

Comment (TedLivist) [reply app/views/devise/sessions/new.html.erb]: Yes. It's not confirmation issue. I fixed it myself, that's why you have the suggestion.

Comment (lily4178993) [reply app/views/devise/sessions/new.html.erb]: Thank you for the feedback. I updated the code as required

**Review by danmainah — APPROVED** [review]
Hi @Ayokunnumi1 and @lily4178993 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the users are able to sign in using email and password.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**